### ISSUE TITLE
Copy files from phone to PC

### ISSUE THREAD

**Issue by rpawar0703** [Genymobile/scrcpy#1227]
Is there any way to copy stuff from phone to pc like drag and drop feature. If not, please implement it. Thanks

**Comment by quancayda** [comment]
Yeah, I also expect scrcpy has that function like way Samsung dex did J.

**Comment by rpawar0703** [comment]
> Yeah, I also expect scrcpy has that function like way Samsung dex did J.

you can use "adb pull sdcard/filename or foldername" to copy files from phone to PC even without scrcpy

**Comment by quancayda** [comment]
> > Yeah, I also expect scrcpy has that function like way Samsung dex did J.
> 
> you can use "adb pull sdcard/filename or foldername" to copy files from phone to PC even without scrcpy

Haha, i mentioned drag copy function to drag a file from scrcpy screen to PC.
No need to take much time for using adb pull while you can copy it via usb cable more quickly :).

**Comment by Nuc1eoN** [comment]
> you can use "adb pull sdcard/filename or foldername" to copy files from phone to PC even without scrcpy

Yes but that is not a visual way of doing it.
Also how would you copy folders from a Work profile using this methoid? It is not possible.

**Comment by necros2k7** [comment]
@Nuc1eoN , @rom1v Simple GUI file manager will be fine like in Myphoneexplorer or like in https://www.tightvnc.com/download.php

**Comment by kagbodji** [comment]
I'll throw this idea in the hat: perhaps a mapped folder where the server and client cooperate to do file copying (perhaps using `adb pull` in the background) could work. Think Rsync between host computer and the phone.
1. when starting scrcpy, provide a mapped dir to the host computer. It could be in the format of `--fle_sync_dir=device_dir:host_dir`. Those directories would be created if they don't exist
2. the user simply copies/pastes or drags/drops files on the `device_dir` to the `host_dir` or vice-versa to see the files appear on either device.
3. the user should be aware that files will be synced between this dirs and it's up to them to manage the dirs (i.e., clearing files as needed -- or there could be a config to auto-remove file from the source after it's been copied over)

**Comment by imvickykumar999** [comment]
This might help, after connecting phone ...

    adb connect 192.168.0.102

Run this file.

```python
from ppadb.client import Client as AdbClient

client = AdbClient(host="127.0.0.1", port=5037)
device = client.device('192.168.0.102:5555')

file='image.png'
path='Download/'
device.pull(f"/sdcard/{path}/{file}", f"static/{file}")
```

Ref. : https://pypi.org/project/pure-python-adb/#pull

**Comment by perrette** [comment]
The idea by @kagbodji does not sound very simple to do in practice. What to do on startup if the folders differ?
I'd imagine instead doing exactly like copy PC -> Phone works: just drop the file in the Download folder from the PC. However, I am not familiar at all with Android, but is it possible to simply get the drag and drop events from any app in USB debug mode? Does this event include the actual path of the underlying data, or (more likely) does it simply contain some visual info? If you can't get the path from any app, then I don't see how that's possible to do that from the phone. The only remaining, and by far simplest option in my limited perspective, is to mount the Android drive on the host computer, and let the user move files around using the usual host computer file manager.

**Comment by furyNerd** [comment]
hi, i'd like to share my solutions for reference.

i use a WinSCP on win and ftp server app on android for accessing files on android from pc, it works though not elegant, especially when too much files to indexing like folders `DCIM/camera` (not sure if it is the problem of the ftp app, sometimes ftp connection failed), and for safety everytime when i need this i must start/stop ftp app on android (with `scrcpy` to start/stop ftp on android is quit easy but not enough)


i think `ftp` could be a good idea for accesssing files on android from pc, but i'm still seeking a method more quick, like start ftp server by `adb cli` (without root), or direct mount android storage on win like 'mtp', and most important, i wish it could work via wifi/hotpot connections between pc and android.
